firmware/common/ui_widget.cpp

```diff
@@ -597,11 +597,14 @@ Console::Console(
 void Console::clear(bool clear_buffer = false) {
 	if(clear_buffer)
 		buffer.clear();
-		
-	display.fill_rectangle(
-		screen_rect(),
-		Color::black()
-	);
+
+	if(!hidden() && visible()){
```

Comment:
what problem you had here? did you tested that this does not affect other apps using the console?

Reply:
I had a problem with using tabs with consoles on both of them. The new one would be showed correctly, but when the old one was hidden, it would call the clear method at some point in the flow, and would clear out portions of the new tab. This makes sure that if clear is called, that the console tab is actually showing. I haven't seen issues with other consoles.